test/connection.test.ts

```diff
@@ -523,6 +523,43 @@ describe('Connection', function () {
       assert.calledOnce(spy)
       assert.calledWith(spy, '58')
     })
+
+    it('should not send money in receiveOnly mode', async function ()  {
+      this.server = await createServer({
+        plugin: this.serverPlugin,
+        serverSecret: Buffer.alloc(32),
+        receiveOnly: true
+      })
+
+      const { destinationAccount, sharedSecret } = this.server.generateAddressAndSecret()
+      this.destinationAccount = destinationAccount
+      this.sharedSecret = sharedSecret
+
+      const connectionPromise = this.server.acceptConnection()
+
+      this.clientConn = await createConnection({
+        plugin: this.clientPlugin,
+        destinationAccount,
+        sharedSecret
+      })
+
+      const moneySpy = sinon.spy()
+      this.clientConn.on('money', moneySpy)
+
+      const dataSpy = sinon.spy()
+      this.clientConn.on('data', dataSpy)
+
+      this.serverConn = await connectionPromise
+      this.serverConn.on('stream', (stream: DataAndMoneyStream) => {
+        stream.setReceiveMax(10000)
+      })
+
+      const clientStream = this.clientConn.createStream()
+      await clientStream.sendTotal(117)
```

Comment:
What is this actually testing, is it testing that a client can send money to the server? 

There are spys on the clientConn 'data' and 'money' events and then a client stream is made and send total is called. If `clientStream.sendTotal` works here then the 'data' and 'money' events would be on the server side of the connection.

Reply:
Yeah, this is incorrect.

The intention is to check that no packets are being sent from the `receiveOnly` node that have money in them. But the money event won't fire on test packets, so this is wrong. (The client stream sending money is unnecessary).

Thinking about this more, I think a spy on `sendData` on the plugin would accomplish what I'm really trying to test -- I'll update it.